### DISCUSSION TITLE
Assistance needed with grbl control box firmware and IoSender

### DISCUSSION THREAD

**Discussion by cfficaurzua** [terjeio/ioSender#285] [General]
Hello,

I've been working on a custom grbl control box firmware, which I built from scratch using an Esp32 ESP-idf. This firmware connects to GrblHAL embedded in an STM32F103C8 through uart1 (rj9 cable). I have implemented a mode for connecting via telnet, where I simply forward the connection between telnet and uart1.

However, when I try to use IoSender in legacy mode, I encounter an issue. A message appears stating, "MPG mode active, waiting for it to be disabled..." I am not sure what this means or how to disable MPG mode.

I would appreciate any assistance or guidance on this matter. 
Thanks

**Comment by terjeio** [comment]
This happens when there is another "sender" requesting real-time reports - when ioSender gets the responses it assumes a MPG is in control and waits for it to relinquish it before continuing with the startup sequence. FYI ioSender reads quite a bit of info on startup - access to this is blocked by the controller when another sender is in control.

More info [here](https://github.com/grblHAL/core/wiki/MPG-and-DRO-interfaces).

**Reply by cfficaurzua** [reply]
Thanks for your quick response. I will block the real-time reports then.
